modelexpress_client/python/modelexpress/client.py

```
@@ -0,0 +1,666 @@
+        url = f"{self.engine_address}/collective_rpc"
+        payload = json.dumps({
+            "method": "start_weight_server",
+            "args": [zmq_address_base],
+        }).encode()
+
+        logger.info(f"Starting weight server via collective_rpc: {url}")
+        start = time.time()
+
+        while (time.time() - start) < timeout_s:
+            try:
+                req = Request(url, data=payload, method="POST")
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)